### PR TITLE
Enable read port from nrepl file

### LINKED ISSUE #5: Add option to read repl port from file
When running repl there is some files that are created with the port of the repl (like `.nrepl-port`). We should allow the user to select in the configuration if they want to use the file instead of a specific port.

### PR DESCRIPTION
**Pull request by afucher**
Closes #5 

Screenshot 2024-01-15 at 09 10 51

### CODE REVIEW

File: src/main/clojure/com/github/clojure_repl/intellij/configuration/repl.clj [hunk 1]
```diff
@@ -42,20 +43,45 @@
   (reset! current-repl* initial-current-repl)
   (swap! db/db* assoc :current-nrepl nil))
 
+(defn is-manual? [editor]
```
Comment (ericdallo) [inline src/main/clojure/com/github/clojure_repl/intellij/configuration/repl.clj:46]:
```suggestion
(defn ^:private manual? [editor]
```

File: src/main/clojure/com/github/clojure_repl/intellij/configuration/repl.clj [hunk 2]
```diff
@@ -96,8 +122,22 @@
     (createConsoleActions [_] (into-array AnAction []))
     (allowHeavyFilters [_])))
 
+(defn ^:private should-read-port-file? []
```
Comment (ericdallo) [inline src/main/clojure/com/github/clojure_repl/intellij/configuration/repl.clj:125]:
```suggestion
(defn ^:private read-port-file? []
```

File: src/main/clojure/com/github/clojure_repl/intellij/configuration/repl.clj [hunk 3]
```diff
@@ -113,9 +153,12 @@
 (defn ^:private apply-editor-to [^RunConfigurationBase configuration-base]
   (let [editor @editor-view*
         host (seesaw/text (seesaw/select editor [:#nrepl-host]))
-        project (seesaw/text (seesaw/select editor [:#project]))]
+        project (seesaw/text (seesaw/select editor [:#project]))
+        mode (if (is-manual? editor) "manual" "repl-file")]
     (swap! db/db* assoc-in [:settings :nrepl-host] host)
     (.setNreplHost (.getOptions configuration-base) host)
+    (swap! db/db* assoc-in [:settings :mode] mode)
```
Comment (ericdallo) [inline src/main/clojure/com/github/clojure_repl/intellij/configuration/repl.clj:160]: `mode` inside whole plugin settings looks like too generic, WDYT about `remote-repl-mode`?

File: src/main/clojure/com/github/clojure_repl/intellij/configuration/repl.clj [hunk 4]
```diff
@@ -96,8 +122,22 @@
...
 (defn ^:private start-process [project]
   (logger/info "Starting NREPL process...")
+  ;TODO: handle when file does not exist
+  (when (should-read-port-file?)
+      (if-let [project (->> (ProjectManager/getInstance)
+                            .getOpenProjects
+                            (filter #(= (.getName %) (.getName project)))
+                            first)]
+        (let [base-path (.getBasePath ^Project project)
+              repl-file (str base-path "/.nrepl-port")
```
Comment (ericdallo) [inline src/main/clojure/com/github/clojure_repl/intellij/configuration/repl.clj:137]: this won't work for windows.
```suggestion
              repl-file (io/file base-path ".nrepl-port")
```

File: src/main/clojure/com/github/clojure_repl/intellij/configuration/repl.clj (hunk 3 above)
Comment (afucher) [reply src/main/clojure/com/github/clojure_repl/intellij/configuration/repl.clj]: @ericdallo do you mean the `:mode` inside `:settings`, or the symbol name?

Comment (ericdallo) [reply src/main/clojure/com/github/clojure_repl/intellij/configuration/repl.clj]: mode inside the settings, i mean, mode for what? the repl? but local or remote? mode sounds a little bit too generic too

File: src/main/clojure/com/github/clojure_repl/intellij/configuration/factory/remote.clj [hunk 5]
```diff
@@ -16,14 +17,29 @@
...
 (set! *warn-on-reflection* true)
 
 (def ^:private options-class ReplRemoteRunOptions)
 
+(defn ^:private read-port-file? []
+  (= (-> @db/db* :settings :mode) :file-config))
+
 (defn ^:private setup-process [project]
   (logger/info "Connecting to nREPL process...")
+  ;TODO: handle when file does not exist
```
Comment (ericdallo) [inline src/main/clojure/com/github/clojure_repl/intellij/configuration/factory/remote.clj:32]: would be nice in the future to ask to user what is the port like cider does

File: src/main/clojure/com/github/clojure_repl/intellij/configuration/factory/remote.clj [hunk 6]
```diff
@@ -16,14 +17,29 @@
...
+  (when (read-port-file?)
+    (if-let [project (->> (ProjectManager/getInstance)
+                          .getOpenProjects
+                          (filter #(= (.getName ^Project %) (.getName ^Project project)))
+                          first)]
```
Comment (ericdallo) [inline src/main/clojure/com/github/clojure_repl/intellij/configuration/factory/remote.clj:37]: Why you are getting the project like that if we have the project as arg [here](https://github.com/afucher/clojure-repl-intellij/pull/41/files#diff-d08de06aff710b6d06d8f6433d70fbf52bd18f225999e32ad345dfe2694a1be1R30)?

File: src/main/clojure/com/github/clojure_repl/intellij/configuration/factory/remote.clj [hunk 7]
```diff
@@ -16,14 +17,29 @@
...
 (def ^:private options-class ReplRemoteRunOptions)
 
+(defn ^:private read-port-file? []
+  (= (-> @db/db* :settings :mode) :file-config))
+
 (defn ^:private setup-process [project]
   (logger/info "Connecting to nREPL process...")
+  ;TODO: handle when file does not exist
+  (when (read-port-file?)
+    (if-let [project (->> (ProjectManager/getInstance)
```
Comment (ericdallo) [inline src/main/clojure/com/github/clojure_repl/intellij/configuration/factory/remote.clj:34]: if it has no else, it should be a `when-let` right? (why are not you using clojure-lsp-intellij 😛 ?)
image

File: src/main/clojure/com/github/clojure_repl/intellij/configuration/factory/remote.clj [hunk 8]
```diff
@@ -32,28 +48,56 @@
...
+                                       :id :repl-file
+                                       :group repl-mode-group
+                                       :mnemonic \R) "wrap"]
+                        [(seesaw/label "Project") ""]
+                        [(seesaw/combobox :id :project
+                                          :model (->> (ProjectManager/getInstance)
+                                                      .getOpenProjects
+                                                      (map #(.getName ^Project %)))) "wrap"]])]
+    (seesaw/listen repl-mode-group :action
+                   (fn [e]
```
Comment (ericdallo) [inline src/main/clojure/com/github/clojure_repl/intellij/configuration/factory/remote.clj:83]:
```suggestion
                   (fn [_e]
```

File: src/main/clojure/com/github/clojure_repl/intellij/configuration/factory/remote.clj (hunk 5 above)
Comment (afucher) [reply src/main/clojure/com/github/clojure_repl/intellij/configuration/factory/remote.clj:32]: I will create a card for that, after merge this PR

**Review by ericdallo — APPROVED** [review]

**Pull request merged**